app/services/rate_creator.rb

```diff
@@ -2,7 +2,8 @@ class RateCreator
   def self.build(value, submission_id, user_id)
     submission = Submission.find(submission_id)
     user = User.find(user_id)
-    new(value, submission, user, RateChecker.new)
+    rate_checker = RateChecker.new(submission.id, user.id)
```

Comment:
I'm not sure if build it here or pass a ready RateChecker object to the `build` method. @lesniakania ?